components/src/core/Drawer/DrawerNavGroup.tsx

```diff
@@ -70,39 +118,67 @@ export type DrawerNavGroupProps = {
     chevron?: boolean;
     content?: ReactNode;
     divider?: boolean;
+    nestedDivider?: boolean;
     fontColor?: string;
     iconColor?: string;
     items: NavItem[];
     onSelect?: Function;
     open?: boolean;
+    ripple?: boolean;
     title?: string;
     titleColor?: string;
+    hidePadding?: boolean;
+    activeItem?: string;
 };
 
-function NavigationListItem(item: NavItem, props: DrawerNavGroupProps): ReactNode {
-    const { title, subtitle, icon, statusColor, onClick, active } = item;
-    const { divider = true } = props;
-    if (!title && !icon) {
-        return null;
-    }
+// renderer function for each nav item / nested nav item
+function NavigationListItem(
+    item: NavItem | NestedNavItem,
+    props: DrawerNavGroupProps,
+    depth = 0,
+    expanded = false
+): ReactNode {
+    const {
+        onClick,
+        title,
+        subtitle,
+        items: subItems,
+        divider: itemDivider,
+        onClickIcon,
+        itemID,
+        chevron: itemChevron,
+    } = item;
+    const icon = (item as NavItem).icon;
+    const { divider: groupDivider = true, nestedDivider } = props;
 
     const classes = useStyles(props);
     const theme = useTheme();
+    // @ts-ignore
+    const primary50Color = theme.palette.primary[50];
     const {
-        // @ts-ignore
-        activeBackgroundColor = theme.palette.type === 'light' ? theme.palette.primary[50] : theme.palette.primary.main,
+        activeBackgroundColor = theme.palette.type === 'light' ? primary50Color : theme.palette.primary.main,
         activeFontColor = theme.palette.type === 'light'
             ? theme.palette.primary.main
             : theme.palette.primary.contrastText,
         activeIconColor = theme.palette.type === 'light'
             ? theme.palette.primary.main
             : theme.palette.primary.contrastText,
-        fontColor,
-        chevron,
+        fontColor = theme.palette.text.secondary,
+        chevron: groupChevron,
         iconColor,
         onSelect,
+        hidePadding,
+        ripple,
+        activeItem,
     } = props;
 
+    let divider;
+    if (depth) {
```

Comment:
```suggestion
    if (depth && depth > 0) {
```

Reply:
`depth` is controlled by us, not by the user. It can only be an non-negative integer. 

`if (depth)` -> if we are not in the root level

Reply:
Just suggesting some ways to make it a little more robust